polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -96,6 +97,9 @@ macro_rules! concat_const {
             cat
         };
 
+        // FIXME eagr: consider opting for `from_utf8_unchecked()`
+        // SAFETY: The concatenation of two string slices is guaranteed to be valid UTF8,
+        // so are the byte slices as they have the same memory layout.
```

Comment:
I would stay with the safe version, so no `unsafe`, since we have a policy around that. (I know we don't always follow it, but we're trying to be better about it going forwards.) It's good to keep the comment but we don't need `SAFETY` anymore:

Also, can we add a doc above `macro_rules! concat_const`?

Reply:
How about this one? (Trades some weird syntax for the absence of runtime code)

```rust
macro_rules! c_c_combo {
    ( $combined:ident : $($name:ident : $type:ty = $value:expr);+ $(;)? ) => {
        $(
            const $name: $type = $value;
        )+
        const $combined: &'static str = concat!($($value),+);
    }
}

c_c_combo! (PATH:
    BASE: &'static str = "path/to";
    PART: &'static str = "foo";
);
```

Reply:
Nice~ But probably not, the NODE_VERSION is not defined in this mod.